reframe/core/schedulers/slurm.py

```diff
@@ -231,19 +231,19 @@ def filternodes(self, job, nodes):
         # Collect options that restrict node selection
         options = job.sched_access + job.options
```

Comment:
It's more correct conceptually if you change just this line:

```suggestion
        # Create a mutable list out of the SequenceView object of `job.sched_access`
        options = list(job.sched_access + job.options)
```